core/arch/arm/plat-rcar/main.c

```diff
@@ -49,6 +60,118 @@ register_dynamic_shm(NSEC_DDR_3_BASE, NSEC_DDR_3_SIZE);
 
 static struct scif_uart_data console_data __nex_bss;
 
+enum rcar_product_type rcar_get_product_type(void)
+{
```

Comment:
The implementation of this function is a bit verbose, but I can't see any way to significantly reduce it.